backend/migrations/2023-03-29-111855_merge_plant_tables/up.sql

```diff
@@ -0,0 +1,8 @@
+ALTER TABLE seeds DROP COLUMN plant_id;
+
+DROP TABLE plants;
+
+ALTER TABLE plant_detail RENAME TO plants;
+ALTER TABLE seeds ADD COLUMN plant_id INTEGER REFERENCES plants(id);
+
+INSERT INTO plants (id, binomial_name, common_name) VALUES (-1, 'Testia testia', ARRAY['Testplant']);
```

Comment:
Test data shouldn't be part of migrations.
Please set up the test data in the test where you need it.
You can use this function to make sure nothing will be committed during a test to ensure test isolation.
https://docs.diesel.rs/master/diesel/connection/trait.Connection.html#method.test_transaction

A quick search led me to this example of how to use it.
https://github.com/diesel-rs/diesel/issues/1549#issuecomment-621792203

Reply:
How are you finding these things when I've been sitting here googling this stuff for hours :sweat_smile:....

Anyways, thanks for the suggestion :+1: .

Reply:
:shrug: 
Here is another example that is closer to our set up.
https://cloudmaker.dev/actix-integration-tests/

But maybe we can leave it for now and create a new issue to set up backend tests in a usable way?

Reply:
Did you already see the tests we already have in backend/src/test/seed.rs?

Btw. @Benjaminimal **excellent** review.

Reply:
I agree there should be further tests, within M1.

Reply:
The seeds tests do the same.
Adding test data in migrations.
https://github.com/ElektraInitiative/PermaplanT/blob/9ac1b16809e94aa9cc6c5d8fadf41f8f74d4c249/backend/migrations/2023-01-21-024920_create_seeds/up.sql#L19-L22

No test transactions in the seeds test.
https://github.com/ElektraInitiative/PermaplanT/blob/9ac1b16809e94aa9cc6c5d8fadf41f8f74d4c249/backend/src/test/seed.rs

We also have a lot of code duplication here at the start of each test function.
This should be pulled out into some setup which also sets up a test database (maybe with some fixtures) so we can use test transactions easily.
We would also need a tear-down to drop this database after the last test has run.
I looked into it briefly and it seems like rust's built-in test crate doesn't provide easy setup and tear-down out of the box.
@kitzbergerg 
Do you know any recipe or good test crate to achieve this?

Reply:
I added a test utility that initializes the pool so that it uses the test transactions.
This should be enough to cover easy test cases.
I made a comment about its limitations which @badnames and I identified.

Reply:
As far as I know the tests in Rust shouldn't actually be used how we use them. They should be more on the side of unit tests rather than integration tests.
I just found a [crate](https://docs.rs/mockall/latest/mockall/) to mock traits (~interfaces). I think we should think about tests more carefully soon (especially as we currently require a running db that is not part of the backend) and split into unit and integration tests.

Reply:
Yes, excellent suggestion, we definitely should separate unit and integration tests. There is #39, can you please kick this off as part of #158?